File: library/src/main/java/moe/leekcake/live2dforjvm/framework/type/id/CubismIdManager.java
package moe.leekcake.live2dforjvm.framework.type.id;

import moe.leekcake.live2dforjvm.framework.jni.id.CubismIdManagerJNI;
import moe.leekcake.live2dforjvm.core.type.AutoPointer;

public class CubismIdManager extends AutoPointer {
    public CubismIdManager(long pointer) {
        super(pointer);
    }

    public void registerIds(String[] ids) {
        CubismIdManagerJNI.RegisterIds(pointer, ids);
    }

    public void registerId(String id) {
        CubismIdManagerJNI.RegisterId(pointer, id);
    }

    public CubismId getId(String id) {
        return new CubismId(CubismIdManagerJNI.GetId(pointer, id));
    }

    public boolean isExist(String id) {
        return CubismIdManagerJNI.IsExist(pointer, id);
    }
}

File: library/src/main/kotlin/moe/leekcake/live2dforjvm/type/CubismIdManager.kt
package moe.leekcake.live2dforjvm.type

import moe.leekcake.live2dforjvm.framework.CubismIdManagerJNI

class CubismIdManager(val pointer: Long) {
    fun registerIds(ids: Array<String>) {
        return CubismIdManagerJNI.CubismIdManager_RegisterIds(pointer, ids)
    }

    fun registerId(id: String) {
        return CubismIdManagerJNI.CubismIdManager_RegisterId(pointer, id)
    }

    fun getId(id: String): CubismId {
        return CubismId(CubismIdManagerJNI.CubismIdManager_GetId(pointer, id))
    }

    fun isExist(id: String): Boolean {
        return CubismIdManagerJNI.CubismIdManager_IsExist(pointer, id)
    }
}